### ISSUE TITLE
Pulls The Full Object?

### ISSUE THREAD

**Issue by JimLynchCodes** [southpolesteve/graphql-cosmos-demo#4]
Hi!

I'm not totally sure if this is possible, but I was hoping that I could use a GraphQL API to pull a subset of a document from cosmosdb, thus reducing my overall RUs.

It looks like this example is querying for the full Todo object.

In my case I have a collection of massive objects, but I only want to pull one field so I'm wondering what the most efficient way to do that would be. :)

Thanks!
Jim

**Comment by southpolesteve** [comment]
@JimLynchCodes You can do that with a query instead of a read. So on this line https://github.com/southpolesteve/graphql-cosmos-demo/blob/master/index.js#L42

You could do something like: container.items.query(`SELECT c.field1, c.field2 FROM c WHERE c.id = ${id}).fetchAll()

**Comment by JimLynchCodes** [comment]
Thanks @southpolesteve 

My team has already selected to use the mongo API so we can't use regular sql... is there any way to do it with the mongo API?

**Comment by southpolesteve** [comment]
If you are using the Mongo API, you want to look into data loader to efficiently fetch the data: https://www.instapainting.com/blog/use-dataloader-to-batch-and-cache-arbitrary-mongodb-queries

**Comment by JimLynchCodes** [comment]
thanks @southpolesteve!

My team and I decided to make regular REST calls with CosmosDB so I kind of forgot I had opened this issue. 😅

We figured out that the regular mongo find query takes a second parameter, "projection" which allows us to pull a subset of an object.

Possibly we could use these projections in the resolvers, although that "dataloader" may be a more elegant / flexible way to do it.

I'll close this issue as I need to spend more time looking into this...

Thanks!
